# Course Project

In [232]:
import pandas as pd
import numpy as np

loandata = pd.read_csv("..//data//lending_club//LoanStats3c_2014.csv")

In [243]:
credit_score_mapping_data = pd.read_csv("..//data//lending_club//credit_score_mapping.csv")
credit_score_mapping_data.head()
print credit_score_mapping_data.dtypes

sub_grade       object
credit_score     int64
dtype: object


In [242]:

emp_yr_mapping_data = pd.read_csv("..//data//lending_club//emp_yr_mapping.csv")
emp_yr_mapping_data.head(10)
print emp_yr_mapping_data.dtypes

emp_length    object
emp_yr         int64
dtype: object


In [19]:
loandata.head()

id   member_id  loan_amnt  funded_amnt  funded_amnt_inv        term  \
0  36805548  39558264.0    10400.0      10400.0          10400.0   36 months   
1  38098114  40860827.0    15000.0      15000.0          15000.0   60 months   
2  37612354  40375473.0    12800.0      12800.0          12800.0   60 months   
3  37822187  40585251.0     9600.0       9600.0           9600.0   36 months   
4  37842129  40605224.0    21425.0      21425.0          21425.0   60 months   

  int_rate  installment grade sub_grade            ...              \
0    6.99%       321.08     A        A3            ...               
1   12.39%       336.64     C        C1            ...               
2   17.14%       319.08     D        D4            ...               
3   13.66%       326.53     C        C3            ...               
4   15.59%       516.36     D        D1            ...               

  num_tl_90g_dpd_24m num_tl_op_past_12m pct_tl_nvr_dlq  percent_bc_gt_75  \
0                0.0                4.0           83.3              14.3   
1                0.0                4.0          100.0               0.0   
2                0.0                0.0           76.9             100.0   
3                0.0                3.0          100.0              60.0   
4                0.0                2.0           91.4             100.0   

  pub_rec_bankruptcies tax_liens tot_hi_cred_lim total_bal_ex_mort  \
0                  0.0       0.0        179407.0           15030.0   
1                  0.0       0.0        196500.0          149140.0   
2                  0.0       0.0        368700.0           18007.0   
3                  0.0       0.0         52490.0           38566.0   
4                  0.0       0.0         57073.0           42315.0   

  total_bc_limit total_il_high_credit_limit  
0        13000.0                    11325.0  
1        10000.0                    12000.0  
2         4400.0                    18000.0  
3        21100.0                    24890.0  
4        15000.0                    35573.0  

[5 rows x 111 columns]

In [20]:
feature_list_all = loandata.columns

In [17]:
feature_list_all

Index([u'id', u'member_id', u'loan_amnt', u'funded_amnt', u'funded_amnt_inv',
       u'term', u'int_rate', u'installment', u'grade', u'sub_grade',
       ...
       u'num_tl_90g_dpd_24m', u'num_tl_op_past_12m', u'pct_tl_nvr_dlq',
       u'percent_bc_gt_75', u'pub_rec_bankruptcies', u'tax_liens',
       u'tot_hi_cred_lim', u'total_bal_ex_mort', u'total_bc_limit',
       u'total_il_high_credit_limit'],
      dtype='object', length=111)

# Feature Selection using business knowledge: 38 out of 111 features are selected using domain knowledge

In [259]:
feature_list_selected = ['id', 'loan_amnt', 'term', 'int_rate', 'installment', 'grade', 'sub_grade',
                         'emp_title', 'emp_length', 'home_ownership', 'annual_inc', 'verification_status',
                         'issue_d', 'loan_status', 'title', 'addr_state', 'dti', 'tax_liens', 'bc_util',
                         'bc_open_to_buy', 'total_bal_ex_mort', 'acc_open_past_24mths', 'chargeoff_within_12_mths',
                         'delinq_2yrs', 'mths_since_last_delinq', 'pct_tl_nvr_dlq', 'num_sats', 'num_tl_120dpd_2m',
                         'num_tl_30dpd', 'num_tl_90g_dpd_24m', 'num_accts_ever_120_pd', 'mths_since_recent_revol_delinq']

In [27]:
loandata_sf = loandata[feature_list_selected]

In [28]:
loandata_sf.head()

id  loan_amnt        term int_rate  installment grade sub_grade  \
0  36805548    10400.0   36 months    6.99%       321.08     A        A3   
1  38098114    15000.0   60 months   12.39%       336.64     C        C1   
2  37612354    12800.0   60 months   17.14%       319.08     D        D4   
3  37822187     9600.0   36 months   13.66%       326.53     C        C3   
4  37842129    21425.0   60 months   15.59%       516.36     D        D1   

                         emp_title emp_length home_ownership  \
0   Truck Driver Delivery Personel    8 years       MORTGAGE   
1                       MANAGEMENT  10+ years           RENT   
2        Senior Sales Professional  10+ years       MORTGAGE   
3                 Admin Specialist  10+ years           RENT   
4  Programming Analysis Supervisor    6 years           RENT   

                ...                chargeoff_within_12_mths delinq_2yrs  \
0               ...                                     0.0         0.0   
1               ...                                     0.0         0.0   
2               ...                                     0.0         1.0   
3               ...                                     0.0         0.0   
4               ...                                     0.0         0.0   

  mths_since_last_delinq pct_tl_nvr_dlq num_sats num_tl_120dpd_2m  \
0                   42.0           83.3     17.0              0.0   
1                    NaN          100.0      6.0              0.0   
2                   17.0           76.9      8.0              0.0   
3                    NaN          100.0     12.0              0.0   
4                   60.0           91.4     10.0              0.0   

   num_tl_30dpd  num_tl_90g_dpd_24m  num_accts_ever_120_pd  \
0           0.0                 0.0                    4.0   
1           0.0                 0.0                    0.0   
2           0.0                 0.0                    1.0   
3           0.0                 0.0                    0.0   
4           0.0                 0.0                    1.0   

   mths_since_recent_revol_delinq  
0                            42.0  
1                             NaN  
2                            17.0  
3                             NaN  
4                            60.0  

[5 rows x 32 columns]

In [29]:
type(loandata_sf)

pandas.core.frame.DataFrame

In [30]:
loandata_sf.shape

(235633, 32)

In [40]:
print loandata_sf.dtypes 

id                                 object
loan_amnt                         float64
term                               object
int_rate                           object
installment                       float64
grade                              object
sub_grade                          object
emp_title                          object
emp_length                         object
home_ownership                     object
annual_inc                        float64
verification_status                object
issue_d                            object
loan_status                        object
title                              object
addr_state                         object
dti                               float64
tax_liens                         float64
bc_util                           float64
bc_open_to_buy                    float64
total_bal_ex_mort                 float64
acc_open_past_24mths              float64
chargeoff_within_12_mths          float64
delinq_2yrs                       

In [54]:
col_list_float = list(loandata_sf.select_dtypes(include=['float64']).columns)
col_list_obj = list(loandata_sf.select_dtypes(include=['object']).columns)   

In [62]:
print  "Float columns:"
print  col_list_float 

print  "\n Object columns:"
print  col_list_obj

float columns
['loan_amnt', 'installment', 'annual_inc', 'dti', 'tax_liens', 'bc_util', 'bc_open_to_buy', 'total_bal_ex_mort', 'acc_open_past_24mths', 'chargeoff_within_12_mths', 'delinq_2yrs', 'mths_since_last_delinq', 'pct_tl_nvr_dlq', 'num_sats', 'num_tl_120dpd_2m', 'num_tl_30dpd', 'num_tl_90g_dpd_24m', 'num_accts_ever_120_pd', 'mths_since_recent_revol_delinq']

 object columns
['id', 'term', 'int_rate', 'grade', 'sub_grade', 'emp_title', 'emp_length', 'home_ownership', 'verification_status', 'issue_d', 'loan_status', 'title', 'addr_state']


In [63]:
rearranged_columns = col_list_float + col_list_obj 

In [64]:
print rearranged_columns

['loan_amnt', 'installment', 'annual_inc', 'dti', 'tax_liens', 'bc_util', 'bc_open_to_buy', 'total_bal_ex_mort', 'acc_open_past_24mths', 'chargeoff_within_12_mths', 'delinq_2yrs', 'mths_since_last_delinq', 'pct_tl_nvr_dlq', 'num_sats', 'num_tl_120dpd_2m', 'num_tl_30dpd', 'num_tl_90g_dpd_24m', 'num_accts_ever_120_pd', 'mths_since_recent_revol_delinq', 'id', 'term', 'int_rate', 'grade', 'sub_grade', 'emp_title', 'emp_length', 'home_ownership', 'verification_status', 'issue_d', 'loan_status', 'title', 'addr_state']


In [225]:
loandata_sf_rearr =loandata_sf[rearranged_columns]
print loandata_sf_rearr.dtypes
loandata_sf_rearr.head()

loan_amnt                         float64
installment                       float64
annual_inc                        float64
dti                               float64
tax_liens                         float64
bc_util                           float64
bc_open_to_buy                    float64
total_bal_ex_mort                 float64
acc_open_past_24mths              float64
chargeoff_within_12_mths          float64
delinq_2yrs                       float64
mths_since_last_delinq            float64
pct_tl_nvr_dlq                    float64
num_sats                          float64
num_tl_120dpd_2m                  float64
num_tl_30dpd                      float64
num_tl_90g_dpd_24m                float64
num_accts_ever_120_pd             float64
mths_since_recent_revol_delinq    float64
id                                 object
term                               object
int_rate                           object
grade                              object
sub_grade                         

loan_amnt  installment  annual_inc    dti  tax_liens  bc_util  \
0    10400.0       321.08     58000.0  14.92        0.0     41.5   
1    15000.0       336.64     78000.0  12.03        0.0      4.7   
2    12800.0       319.08    125000.0   8.31        0.0    103.2   
3     9600.0       326.53     69000.0  25.81        0.0     69.2   
4    21425.0       516.36     63800.0  18.49        0.0     97.8   

   bc_open_to_buy  total_bal_ex_mort  acc_open_past_24mths  \
0          7599.0            15030.0                   7.0   
1          9525.0           149140.0                   5.0   
2             0.0            18007.0                   2.0   
3          6494.0            38566.0                   8.0   
4           324.0            42315.0                   4.0   

   chargeoff_within_12_mths    ...      grade  sub_grade  \
0                       0.0    ...          A         A3   
1                       0.0    ...          C         C1   
2                       0.0    ...          D         D4   
3                       0.0    ...          C         C3   
4                       0.0    ...          D         D1   

                         emp_title  emp_length  home_ownership  \
0   Truck Driver Delivery Personel     8 years        MORTGAGE   
1                       MANAGEMENT   10+ years            RENT   
2        Senior Sales Professional   10+ years        MORTGAGE   
3                 Admin Specialist   10+ years            RENT   
4  Programming Analysis Supervisor     6 years            RENT   

   verification_status  issue_d  loan_status                    title  \
0         Not Verified   14-Dec      Current  Credit card refinancing   
1      Source Verified   14-Dec      Current       Debt consolidation   
2             Verified   14-Dec      Current            Car financing   
3      Source Verified   14-Dec   Fully Paid       Debt consolidation   
4      Source Verified   14-Dec      Current  Credit card refinancing   

  addr_state  
0         CA  
1         VA  
2         CA  
3         NJ  
4         MO  

[5 rows x 32 columns]

In [ ]:
# Lets convert
'''
term                               object
int_rate                           object
grade                              object
sub_grade                          object
emp_length                         object
issue_d                            object
loan_status                        object
'''

In [227]:
loandata_sf_obj = loandata_sf[col_list_obj]
loandata_sf_obj.head() 
loandata_sf_obj.term.unique()

term_grp = loandata_sf_rearr.groupby("term")["id"]
print term_grp.size()


# Converting term to numeric variable
loandata_sf_rearr['term'] = loandata_sf_rearr.term.str[0:3]
loandata_sf_rearr['term'] 

term
 36    162570
 60     73059
dtype: int64


0          36
1          60
2          60
3          36
4          60
5          36
6          36
7          36
8          36
9          36
10         60
11         36
12         36
13         60
14         36
15         36
16         36
17         36
18         36
19         60
20         60
21         36
22         36
23         36
24         36
25         60
26         36
27         36
28         36
29         36
         ... 
235603     36
235604     36
235605     36
235606     36
235607     36
235608     36
235609     36
235610     36
235611     36
235612     36
235613     36
235614     60
235615     36
235616     36
235617     60
235618     36
235619     60
235620     36
235621     36
235622     36
235623     36
235624     60
235625     60
235626     60
235627     36
235628     36
235629    NaN
235630    NaN
235631    NaN
235632    NaN
Name: term, dtype: object

In [228]:
# loandata_sf_obj.convert_objects(convert_numeric=True)

loandata_sf_rearr['int_rate'] = loandata_sf_rearr['int_rate'].replace('%','',regex=True).astype('float')/100
loandata_sf_rearr.int_rate.dtype
loandata_sf_rearr.head()


loan_amnt  installment  annual_inc    dti  tax_liens  bc_util  \
0    10400.0       321.08     58000.0  14.92        0.0     41.5   
1    15000.0       336.64     78000.0  12.03        0.0      4.7   
2    12800.0       319.08    125000.0   8.31        0.0    103.2   
3     9600.0       326.53     69000.0  25.81        0.0     69.2   
4    21425.0       516.36     63800.0  18.49        0.0     97.8   

   bc_open_to_buy  total_bal_ex_mort  acc_open_past_24mths  \
0          7599.0            15030.0                   7.0   
1          9525.0           149140.0                   5.0   
2             0.0            18007.0                   2.0   
3          6494.0            38566.0                   8.0   
4           324.0            42315.0                   4.0   

   chargeoff_within_12_mths    ...      grade  sub_grade  \
0                       0.0    ...          A         A3   
1                       0.0    ...          C         C1   
2                       0.0    ...          D         D4   
3                       0.0    ...          C         C3   
4                       0.0    ...          D         D1   

                         emp_title  emp_length  home_ownership  \
0   Truck Driver Delivery Personel     8 years        MORTGAGE   
1                       MANAGEMENT   10+ years            RENT   
2        Senior Sales Professional   10+ years        MORTGAGE   
3                 Admin Specialist   10+ years            RENT   
4  Programming Analysis Supervisor     6 years            RENT   

   verification_status  issue_d  loan_status                    title  \
0         Not Verified   14-Dec      Current  Credit card refinancing   
1      Source Verified   14-Dec      Current       Debt consolidation   
2             Verified   14-Dec      Current            Car financing   
3      Source Verified   14-Dec   Fully Paid       Debt consolidation   
4      Source Verified   14-Dec      Current  Credit card refinancing   

  addr_state  
0         CA  
1         VA  
2         CA  
3         NJ  
4         MO  

[5 rows x 32 columns]

In [194]:
# Converting sub_grade_grp to numeric variable

sub_grade_grp = loandata_sf_obj.groupby("sub_grade")
# print sub_grade_grp.int_rate.mean()
sub_grade_grp_df = pd.DataFrame(sub_grade_grp.size())
type(sub_grade_grp_df)
sub_grade_grp_df

pandas.core.frame.DataFrame

In [250]:
loandata_sf_final = loandata_sf_rearr.merge(credit_score_mapping_data, on = "sub_grade")


In [213]:
# Converting emp_length to numeric variable

print loandata_sf_obj.emp_length.unique()
emp_length_grp = loandata_sf_obj.groupby('emp_length')
print emp_length_grp.size()

['8 years' '10+ years' '6 years' '< 1 year' '2 years' '9 years' '7 years'
 '1 year' '3 years' '5 years' 'n/a' '4 years' nan]
emp_length
1 year       14593
10+ years    79505
2 years      20487
3 years      18267
4 years      13528
5 years      13051
6 years      11821
7 years      13099
8 years      11853
9 years       9424
< 1 year     17982
n/a          12019
dtype: int64


In [251]:
loandata_sf_final = loandata_sf_final.merge(emp_yr_mapping_data, on = "emp_length")
loandata_sf_final.tail(50)

loan_amnt  installment  annual_inc    dti  tax_liens  bc_util  \
235579    19375.0       568.57     54000.0  21.22        0.0     99.4   
235580    24650.0       723.37     72000.0  28.65        0.0     30.0   
235581    12850.0       377.09     60000.0  10.21        0.0     89.5   
235582    25000.0       733.64     67000.0   1.83        0.0     60.4   
235583    20000.0       586.91     65497.0  22.48        0.0    100.0   
235584    10000.0       293.46     95000.0  13.40        2.0     21.9   
235585    12875.0       377.83     61000.0  20.34        0.0    102.0   
235586    18550.0       544.36     72000.0   7.67        0.0     48.7   
235587    30000.0       898.04     85000.0   3.04        0.0     64.0   
235588    11875.0       355.48     29675.0  21.40        0.0     42.3   
235589    35000.0      1047.71    300000.0  13.18        0.0     78.6   
235590    34875.0      1404.95     90000.0  28.01        0.0     98.3   
235591    10275.0       307.58     74000.0  11.79        0.0     69.3   
235592    33150.0       992.34    100000.0  23.80        0.0     94.3   
235593    35000.0      1047.71    114000.0  26.56        0.0     97.8   
235594    10000.0       299.35    125000.0   6.13        0.0     23.7   
235595    35000.0      1039.03    160000.0  10.75        0.0     56.8   
235596    12175.0       361.44     34000.0  26.89        0.0     52.3   
235597    18000.0       534.36     50000.0  17.71        0.0     76.5   
235598    14375.0       575.89     45000.0  21.29        1.0     58.6   
235599    25000.0       742.17    200000.0  19.65        0.0     70.6   
235600     9475.0       379.59     46800.0   3.28        0.0     56.7   
235601    10000.0       296.87     40000.0  24.90        1.0     56.6   
235602    10850.0       322.10     26000.0  34.95        0.0     89.3   
235603    10000.0       296.87    125000.0   8.81        0.0     73.9   
235604    28000.0       831.23     75000.0   8.70        0.0     93.5   
235605    14700.0       436.40     41000.0  37.21        2.0     95.8   
235606    21000.0       623.42     75000.0  22.32        0.0     96.8   
235607    20725.0       615.26     65000.0  27.50        0.0     98.0   
235608    11825.0       351.05     55000.0   7.63        0.0    101.0   
235609     4800.0       192.30     47000.0  14.76        0.0    100.0   
235610     9000.0       360.56     30000.0  28.60        0.0     99.1   
235611     9600.0       384.60     53151.0  18.56        0.0     95.5   
235612    30000.0       890.60    293000.0  11.41        0.0     83.2   
235613    10975.0       325.81     60000.0   1.88        0.0     59.4   
235614    20400.0       605.61     60000.0  28.12        0.0     56.7   
235615    35000.0      1039.03    160000.0  12.37        1.0     54.5   
235616    18000.0       534.36     92000.0  28.08        0.0     97.1   
235617    30000.0       890.60    108000.0  30.80        0.0     89.9   
235618    31975.0       949.23     80000.0  12.57        0.0     99.5   
235619     9175.0       367.57     65000.0  13.65        0.0     71.3   
235620    18000.0       539.57     53000.0  15.76        0.0     53.1   
235621    34875.0      1045.42    120000.0  17.74        0.0     84.2   
235622    10000.0       299.77    141000.0  26.89        0.0     50.6   
235623    34550.0      1035.68     96000.0  12.35        0.0     17.5   
235624    20950.0       628.00     50000.0  11.38        0.0     44.1   
235625    30000.0       899.29    110000.0  28.02        0.0    100.3   
235626    14200.0       425.66     84000.0  23.96        0.0     92.0   
235627    19525.0       585.29    128749.0  17.85        0.0     92.5   
235628    26400.0       791.37     69745.8  28.89        0.0    101.7   

        bc_open_to_buy  total_bal_ex_mort  acc_open_past_24mths  \
235579            89.0            31764.0                   4.0   
235580         28981.0            48731.0                   5.0   
235581           538.0            14340.0                   3.0   
235582          

In [214]:
# loandata_sf_obj['emp_yrs'] = loandata_sf_obj['emp_length'].str.extract('(\d\d)')
# loandata_sf_obj
# this is not working

In [219]:
# loandata_sf_obj['issue_d_formatted'] = pd.to_datetime(loandata_sf_obj.issue_d,  infer_datetime_format)

In [145]:
dlq_status_grp = loandata_sf_final.groupby("loan_status")["id"]
print dlq_status_grp.size()

loan_status
Charged Off            17676
Current               145700
Default                  427
Fully Paid             65090
In Grace Period         1705
Late (16-30 days)        892
Late (31-120 days)      4139
dtype: int64


In [252]:
def dlq_combined(row):
    if row["loan_status"] == 'Current':
        return  0
    elif row["loan_status"] == 'Fully Paid':
        return  0
    else:
        return 1

In [111]:
test_list = [ 'id', 'loan_status']
test_data = loandata_sf_rearr[test_list]
test_data.head()

id loan_status
0  36805548     Current
1  38098114     Current
2  37612354     Current
3  37822187  Fully Paid
4  37842129     Current

In [135]:
test_data['dlq_status'] = new_data.apply(dlq_combined, axis=1)
test_data.head()

C:\Anaconda2\lib\site-packages\ipykernel\__main__.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  if __name__ == '__main__':


id loan_status  dlq_status
0  36805548     Current           0
1  38098114     Current           0
2  37612354     Current           0
3  37822187  Fully Paid           0
4  37842129     Current           0

In [253]:
# loandata_sf_rearr['dlq_status'] = loandata_sf_final.apply(lambda row: dlq_combined(row), axis=1)

loandata_sf_final['dlq_status'] = loandata_sf_final.apply(dlq_combined, axis=1)

In [254]:
loandata_sf_final

loan_amnt  installment  annual_inc    dti  tax_liens  bc_util  \
0         10400.0       321.08     58000.0  14.92        0.0     41.5   
1         15000.0       463.09     80000.0  14.39        0.0     79.2   
2         28000.0       864.44    130000.0   8.61        0.0     29.1   
3         12000.0       370.48     70000.0  21.33        0.0     56.8   
4         24000.0       740.95    120000.0  16.57        0.0     47.1   
5         28000.0       864.44    359564.0   8.08        0.0     64.0   
6         12675.0       391.31     83000.0  29.75        0.0     96.7   
7          6000.0       185.24     99500.0  22.86        0.0     54.0   
8         20000.0       617.46    265000.0   6.58        0.0     41.1   
9          2100.0        64.84     72000.0   6.66        0.0     31.9   
10        18000.0       555.71    184440.0  10.87        0.0     81.7   
11        10000.0       308.73     50000.0  11.79        0.0     80.6   
12        10500.0       324.17     60000.0  11.56        0.0     31.1   
13        13000.0       401.35     89505.0   7.12        0.0     44.4   
14        12000.0       370.48     34000.0  13.55        0.0     96.5   
15        19000.0       586.58     82500.0  10.01        0.0     23.5   
16         5000.0       154.37     25000.0   7.97        0.0     73.1   
17         8500.0       262.42     40000.0   8.40        0.0     76.6   
18        24000.0       740.95     62350.0  17.46        0.0     75.5   
19        23000.0       710.07     75000.0  13.78        0.0      0.0   
20        23000.0       710.07     55000.0  17.08        0.0     37.7   
21        16000.0       493.97     70000.0  17.79        0.0     31.1   
22         5000.0       154.37     70000.0  15.24        0.0     17.4   
23        10000.0       308.73     30000.0  21.52        0.0      9.0   
24        10000.0       308.73     60000.0  17.43        0.0     26.4   
25        35000.0      1080.54    130000.0  17.75        0.0     39.6   
26         6000.0       185.24     85000.0  17.80        0.0     67.1   
27        13000.0       401.35     82530.0  14.16        0.0     75.1   
28        15000.0       463.09     85000.0  25.50        0.0     84.3   
29         2400.0        74.10     70000.0   9.24        0.0      0.0   
...           ...          ...         ...    ...        ...      ...   
235599    25000.0       742.17    200000.0  19.65        0.0     70.6   
235600     9475.0       379.59     46800.0   3.28        0.0     56.7   
235601    10000.0       296.87     40000.0  24.90        1.0     56.6   
235602    10850.0       322.10     26000.0  34.95        0.0     89.3   
235603    10000.0       296.87    125000.0   8.81        0.0     73.9   
235604    28000.0       831.23     75000.0   8.70        0.0     93.5   
235605    14700.0       436.40     41000.0  37.21        2.0     95.8   
235606    21000.0       623.42     75000.0  22.32        0.0     96.8   
235607    20725.0       615.26     65000.0  27.50        0.0     98.0   
235608    11825.0       351.05     55000.0   7.63        0.0    101.0   
235609     4800.0       192.30     47000.0  14.76        0.0    100.0   
235610     9000.0       360.56     30000.0  28.60        0.0     99.1   
235611     9600.0       384.60     53151.0  18.56        0.0     95.5   
235612    30000.0       890.60    293000.0  11.41        0.0     83.2   
235613    10975.0       325.81     60000.0   1.88        0.0     59.4   
235614    20400.0       605.61     60000.0  28.12        0.0     56.7   
235615    35000.0      1039.03    160000.0  12.37        1.0     54.5   
235616    18000.0       534.36     92000.0  28.08        0.0     97.1   
235617    30000.0       890.60    108000.0  30.80        0.0     89.9   
235618    31975.0       949.23     80000.0  12.57        0.0     99.5   
235619     9175.0       367.57     65000.0  13.65        0.0     71.3   
235620    18000.0       539.57     53000.0  15.76        0.0     53.1   
235621    34875.0      1045.42    120000.0  17.74        0.0     8

In [256]:
loandata_sf_final.credit_score.corr(loandata_sf_final.int_rate, method="spearman")

-0.99250545617773756

In [257]:
loandata_sf_final.credit_score.corr(loandata_sf_final.dlq_status, method="spearman")

-0.18132396314637061

In [258]:
loandata_sf_final.to_csv("..//data//lending_club//loandata_sf_final.csv")
